File: day-3/solution.cpp
class Solution {
public:

int BonusPoints(vector<int>& rankings) {
	int n=rankings.size();
	vector<int>points(n,1);
	for(int i=1;i<n;i++){
    	if(rankings[i]>rankings[i-1])
               points[i]=points[i-1]+1;
	}
	int sum=points[n-1];
	for(int i=n-2;i>=0;i--){
    	if(rankings[i]>rankings[i+1])
        	points[i]=max(points[i],points[i+1]+1);
    	sum+=points[i];
	}
   return sum;
}

};

File: day-3/solution.py
class Solution:
   
	def BonusPoints(self, rankings):
    	if not rankings:
        	return 0

    	n = len(rankings)
    	points = [1] * n
    	for i in range(1, n):
        	if rankings[i] > ratings[i - 1]:
            	points[i] = points[i - 1] + 1
       	 
    	for i in range(n - 2, -1, -1):
        	if rankings[i] > rankings[i + 1] and points[i] <= points[i + 1]:
            	points[i] = points[i + 1] + 1

    	return sum(points)
